src/ripple/app/paths/tests/Path_test.cpp

```diff
@@ -1,212 +1,585 @@
+        env.require(balance("bob", USD(70)));
+        env.require(balance("bob", gw2_USD(7)));
+        env.require(balance(gw, Account("alice")["USD"](0)));
+        env.require(balance(gw, Account("bob")["USD"](-70)));
+        env.require(balance(gw2, Account("alice")["USD"](-62.3)));
+        env.require(balance(gw2, Account("bob")["USD"](-7)));
+    }
 
-        // Set credit limits
-        trust(alice, mtgox, "USD", 600, ledger);
-        trust(bob, mtgox, "USD", 700, ledger);
+    void
+    test_alternative_paths_limit_returned_paths_to_best_quality()
+    {
+        testcase("alternative paths - limit returned paths to best quality");
+        Env env(*this);
+        auto const gw = Account("gateway");
+        auto const USD = gw["USD"];
+        auto const gw2 = Account("gateway2");
+        auto const gw2_USD = gw2["USD"];
+        env.fund(XRP(10000), "alice", "bob", "carol", "dan", gw, gw2);
+        env(rate("carol", 1.1));
+        env.trust(Account("carol")["USD"](800), "alice", "bob");
+        env.trust(Account("dan")["USD"](800), "alice", "bob");
+        env.trust(USD(800), "alice", "bob");
+        env.trust(gw2_USD(800), "alice", "bob");
+        env.trust(Account("alice")["USD"](800), "dan");
+        env.trust(Account("bob")["USD"](800), "dan");
+        env(pay(gw2, "alice", gw2_USD(100)));
+        env(pay("carol", "alice", Account("carol")["USD"](100)));
+        env(pay(gw, "alice", USD(100)));
+
+        auto const alternatives = findPath(env.ledger, "alice", "bob",
+            {USD}, Account("bob")["USD"](5));
+        Json::Value jv;
+        Json::Reader().parse(R"([{
+		        "paths_canonical" : [],
+		        "paths_computed" : [],
+		        "source_amount" :
+		        {
+			        "currency" : "USD",
+			        "issuer" : "r9QxhA9RghPZBbUchA9HkrmLKaWvkLXU29",
+			        "value" : "5"
+		        }
+            }])", jv);
+        expect(jv == alternatives);
+    }
 
-        // Distribute funds.
-        pay(mtgox, alice, "USD", "70", ledger);
+    void
+    test_issues_path_negative_issue()
+    {
+        testcase("path negative: Issue #5");
+        Env env(*this);
+        env.fund(XRP(10000), "alice", "bob", "carol", "dan");
+        env.trust(Account("bob")["USD"](100), "alice", "carol", "dan");
+        env.trust(Account("alice")["USD"](100), "dan");
+        env.trust(Account("carol")["USD"](100), "dan");
+        env(pay("bob", "carol", Account("bob")["USD"](75)));
+        env.require(balance("bob", Account("carol")["USD"](-75)));
+        env.require(balance("carol", Account("bob")["USD"](75)));
+
+        auto alternatives = findPath(env.ledger, "alice", "bob",
+            {Account("alice")["USD"]}, Account("bob")["USD"](25));
+        expect(alternatives.size() == 0);
+
+        alternatives = findPath(env.ledger, "alice", "bob",
+            {Account("alice")["USD"]}, Account("alice")["USD"](25));
+        expect(alternatives.size() == 0);
+
```

Comment:
The original test also attempted a payment, and expecting it to fail.
```
              self.what = "alice fails to send to bob.";

              $.remote.transaction()
                .payment('alice', 'bob', "25/USD/alice")
                .once('submitted', function (m) {
                    // console.log("proposed: %s", JSON.stringify(m));
                    callback(m.engine_result !== 'tecPATH_DRY');
                  })
                .submit();
```
You could do that here, too with something like
```env(pay("alice", "bob", Account("alice")["USD"](25)), tec(tecPATH_DRY));```